### PR TITLE
[Testcase Refactoring] Make test/test_module_tracker.py device-agnostic

### PR DESCRIPTION
**Pull request by tgrhn**
## Summary

Register `TestModuleTracker` (test/test_module_tracker.py) with
`instantiate_device_type_tests` instead of running as a plain CPU-only
`unittest.TestCase`, so the suite exercises
`torch.utils.module_tracker.ModuleTracker`'s hook-based forward/backward
hierarchy tracking on every available device type — including MPS and XPU —
and tags the class so it participates correctly in `--hw-classification`
filtering (#186918).

## Changes

**`test/test_module_tracker.py`**:
- Added a `device` parameter to `test_module_hierarchy`,
  `test_confused_hierarchy`, and `test_bw_detection`
- Threaded `device=device` through tensor construction and moved the
  `nn.Module` instances (`Mod`, `MyMod`, `nn.Linear`) onto `device` via
  `.to(device)`
- Registered the class with `instantiate_device_type_tests(TestModuleTracker,
  globals(), allow_mps=True, allow_xpu=True)`, replacing implicit CPU-only
  execution
- Added `hw_classification = HardwareClassification.GENERIC` to
  `TestModuleTracker`
- No assertion values or control flow changed

## Motivation

`ModuleTracker`'s hook-registration/ordering logic is pure Python
bookkeeping — every assertion is on a `set`/`list`/`bool` value describing
which modules ran and in what order, not on tensor numerics — so there's no
reason it should be restricted to a single implicit device. The file had no
device-generic test infrastructure at all before this change.

`hw_classification = HardwareClassification.GENERIC` follows #186918's
`HardwareClassification` enum (added to `common_utils.py` so test classes can
opt into `--hw-classification` filtering).

## Test Plan

    python test/test_module_tracker.py -v
    python test/test_module_tracker.py --hw-classification GENERIC -v
    python test/test_module_tracker.py --hw-classification ACCELERATOR -v

6/6 tests pass (CPU + MPS).

### CODE REVIEW

File: test/test_module_tracker.py [hunk 1]
```diff
@@ -17,7 +18,7 @@
 class TestModuleTracker(TestCase):
```
Comment (Niran814804102) [inline test/test_module_tracker.py:18]: Please add the label `hw_classification = HardwareClassification`, eg. hw_classification = HardwareClassification.ACCELERATOR

**Review by Niran814804102 — APPROVED** [review]
LGTM

**Review by fffrog — CHANGES_REQUESTED** [review]
Hi @tgrhn, thank you for contributing this file.

From my perspective, I think the `ModuleTracker` is not strongly related to any accelerator, which is shared across all the accelerator and do NOT have any specific-accelerator code logic, so I think keeping it as `GENERIC` should be a better option.

Therefore, could you mind restoring all the changes you given and just add the `GENERIC` tag for it, so sorry for the inconvenience.

**Comment by tgrhn** [comment]
> Hi @tgrhn, thank you for contributing this file.
> 
> From my perspective, I think the `ModuleTracker` is not strongly related to any accelerator, which is shared across all the accelerator and do NOT have any specific-accelerator code logic, so I think keeping it as `GENERIC` should be a better option.
> 
> Therefore, could you mind restoring all the changes you given and just add the `GENERIC` tag for it, so sorry for the inconvenience.

Hi @fffrog , thanks for the feedback! You're right that ModuleTracker's hook-ordering
logic isn't accelerator-specific. I've reverted the
instantiate_device_type_tests/device-parametrization changes and just
added hw_classification = HardwareClassification.GENERIC instead. Pushed.

**Review by Niran814804102 — APPROVED** [review]

**Review by fffrog — APPROVED** [review]
SGTM, thank you.

**Review by albanD — APPROVED** [review]
Ok
